parquet/src/main/java/org/apache/iceberg/parquet/ParquetValueWriters.java

```diff
@@ -324,6 +358,33 @@ public void write(int repetitionLevel, T value) {
     public void setColumnStore(ColumnWriteStore columnStore) {
       writer.setColumnStore(columnStore);
     }
+
+    @Override
+    public Stream<FieldMetrics<?>> metrics() {
+      if (writer instanceof PrimitiveWriter) {
+        List<FieldMetrics<?>> fieldMetricsFromWriter = writer.metrics().collect(Collectors.toList());
+
+        if (fieldMetricsFromWriter.size() == 0) {
+          // we are not tracking field metrics for this type ourselves
+          return Stream.empty();
+        } else if (fieldMetricsFromWriter.size() == 1) {
+          FieldMetrics<?> metrics = fieldMetricsFromWriter.get(0);
+          return Stream.of(
+              new FieldMetrics<>(metrics.id(),
+                  metrics.valueCount() + nullValueCount, nullValueCount,
+                  metrics.nanValueCount(), metrics.lowerBound(), metrics.upperBound())
+          );
+        } else {
+          throw new IllegalStateException(String.format(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: &#39;size&#39; expression. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 378. Consider formatting this data with &#39;%d&#39; instead.